app/src/docs/components/CanvasModuleHelp/index.jsx

```diff
@@ -17,16 +17,35 @@ type PortHelpProps = {
 // doesn't play well with inline elements.
 const capitalizeText = (text: string): string => text.charAt(0).toUpperCase() + text.slice(1)
 
+const hasTrailingFullStop = (text: string): string => {
+    if (text.slice(-1) === '.') {
+        return true
+    }
+    return false
+}
+
 function PortHelp({ help, port }: PortHelpProps) {
+    const portName = capitalizeText(port.displayName || port.name)
+
     return (
         <div className={styles.portHelp}>
-            <span className={styles.portName} title={port.name}>{port.displayName || port.name}</span>
-            {port.type.split(/\s+/).map((type) => (
-                <span className={styles.portType} key={type}> {type}.</span>
-            ))}
-            {!help ? null : (<span className={styles.portText}> <ReactMarkdown source={capitalizeText(help) || ''} />.</span>)}
+            <span className={styles.portName} title={port.name}>{portName}</span>
+            {port.type.split(/\s+/).map((type) => {
+                const portType = capitalizeText(type)
+                return (
+                    <span className={styles.portType} key={type}>
+                        {portType === portName && help ? '.' : ` ${type}${hasTrailingFullStop(portType) ? null : '.'}`}
+                    </span>
+                )
+            })}
+            {!help ? null
+                : (
+                    <span className={styles.portText}> <ReactMarkdown source={capitalizeText(help) || ''} />
+                        {hasTrailingFullStop(help) ? null : '.'}
```

Comment:
hm, I wonder if it'd be better to transform the original text, might be cases where an auto-injected `.` doesn't work e.g. if the help text ends with a link or some other markup 

e.g. 
[link.]()
would become:
[link.]().